File: duplicate_detection_using_a_set/dduas.ts
function find_duplicates<T>(elements: string[]): string[] { // T[] whatever type array contains 
    const seen = new Set<string>(); // or use String[]
    const duplicates = new Set<string>();

    for (const item of elements) {
        if (seen.has(item)) {
            duplicates.add(item); 
        } else {
            seen.add(item);
            }
        }

    return Array.from(duplicates)
}

File: duplicate_detection_using_a_set/dduas.java
import java.util.*;

class Solution { // similar to TypeScript
    public <T> List<T> findDuplicates(List<T> elements) {
        Set<T> seen = new HashSet<>();
        Set<T> duplicates = new HashSet<>();

        for (T item : elements) {
            if (seen.contains(item)) {
                duplicates.add(item);
            } else {
                seen.add(item);
            }
        }

        return new ArrayList<>(duplicates);
    }
}
